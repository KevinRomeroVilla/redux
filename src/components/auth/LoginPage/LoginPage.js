import React from "react";
import { useLocation, useNavigate } from "react-router-dom";

import LoginForm from "./LoginForm";
//import useMutation from "../../../hooks/useMutation";
import { useDispatch, useSelector } from "react-redux";
import {
  authLogin,
  uiResetError,
} from "../../../store/Action_Creators/actions";
import { getUi } from "../../../store/selectors";

function LoginPage() {
  const navigate = useNavigate();
  const location = useLocation();
  const dispatch = useDispatch();
  const { isLoading, error } = useSelector(getUi);
  const handleResetError = () => dispatch(uiResetError());

  const handleSubmit = async (credentials) => {
    //event.preventDefault();
    debugger;
    await dispatch(authLogin(credentials)).then(() => {
      const from = location.state?.from?.pathname || "/";
      navigate(from, { replace: true });
    });

    // execute(credentials)
    //   .then(dispatch(authLoginSuccess()))
    //   .then(() => {
    //     const from = location.state?.from?.pathname || "/";
    //     navigate(from, { replace: true });
    //   });
  };

  return (
    <div>
      <LoginForm onSubmit={handleSubmit} isLoading={isLoading} />
      {isLoading && <p>...login in nodepop</p>}
      {error && (
        <div onClick={handleResetError} style={{ color: "red" }}>
          {error.message}
        </div>
      )}
    </div>
  );
}

export default LoginPage;
