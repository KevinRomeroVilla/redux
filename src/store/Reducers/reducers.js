import {
  ADVERT_CREATED,
  ADVERT_LOADED,
  AUTH_LOGIN,
  AUTH_LOGOUT,
} from "../types/types";

const defaultState = {
  auth: false,
  adverts: [],
};

export default function reducer(state = defaultState, action) {
  switch (action.type) {
    case AUTH_LOGIN:
      return { ...state, auth: true };
    case AUTH_LOGOUT:
      return { ...state, auth: false };
    case ADVERT_LOADED:
      return { ...state, adverts: action.payload };
    case ADVERT_CREATED:
      return;

    default:
      return state;
  }
}
