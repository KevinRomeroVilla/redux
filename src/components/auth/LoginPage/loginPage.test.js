import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { defaultState } from "../../../store/Reducers/reducers";
import LoginPage from "./LoginPage";

describe("LoginPage", () => {
  test("Snapshot", () => {
    const store = {
      getState: () => defaultState,
      dispatch: () => {},
      subscribe: () => {},
    };
    const { container } = render(
      <Provider store={store}>
        <LoginPage />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
