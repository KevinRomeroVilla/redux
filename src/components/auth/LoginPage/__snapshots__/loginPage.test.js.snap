// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage Snapshot 1`] = `
<div>
  <div>
    <form>
      <input
        name="email"
        value=""
      />
      <input
        name="password"
        type="password"
        value=""
      />
      <input
        name="remember"
        type="checkbox"
      />
      <button
        disabled=""
      >
        Login
      </button>
    </form>
  </div>
</div>
`;
